### DISCUSSION TITLE
Flixel debugger "Stats" window broken?

### DISCUSSION THREAD

**Discussion by Lbones-ai** [HaxeFlixel/flixel#2322] [Q&A] [answered]
flixel: 4.6.2 4.6.3 4.8.0 4.8.1 [4.9.0]
hscript: 2.3.0 [2.4.0]
lime: 3.0.1 7.5.0 7.7.0 [7.9.0]
openfl: 4.0.1 8.9.1 8.9.6 [9.1.0]

I just updated haxe 4.2.2 and all my installed haxelibs, now when I open the flixel debugger the text in "Stats" is broken:
![immagine](https://user-images.githubusercontent.com/84868536/122238125-2d100780-cec0-11eb-84bd-e5e6e28175f3.png)

Also I'm getting this warnings:
E:/HaxeToolkit/haxe/lib/hscript/2,4,0/hscript/Interp.hx:145: characters 8-18 : Warning : Std.is is deprecated. Use Std.isOfType instead.
E:/HaxeToolkit/haxe/lib/hscript/2,4,0/hscript/Interp.hx:180: characters 8-18 : Warning : Std.is is deprecated. Use Std.isOfType instead.
E:/HaxeToolkit/haxe/lib/hscript/2,4,0/hscript/Interp.hx:221: characters 8-18 : Warning : Std.is is deprecated. Use Std.isOfType instead.
E:/HaxeToolkit/haxe/lib/hscript/2,4,0/hscript/Interp.hx:515: characters 8-18 : Warning : Std.is is deprecated. Use Std.isOfType instead.
E:/HaxeToolkit/haxe/lib/hscript/2,4,0/hscript/Interp.hx:483: characters 37-56 : Warning : Std.is is deprecated. Use Std.isOfType instead.
E:/HaxeToolkit/haxe/lib/hscript/2,4,0/hscript/Interp.hx:484: characters 31-47 : Warning : Std.is is deprecated. Use Std.isOfType instead.
E:/HaxeToolkit/haxe/lib/hscript/2,4,0/hscript/Interp.hx:646: characters 10-42 : Warning : Std.is is deprecated. Use Std.isOfType instead.

**Comment by larsiusprime** [comment]
Can you confirm you're using the hl target?
I've seen this before too, it doesn't happen in cpp or html5 targets.

On Wed, Jun 16, 2021 at 9:31 AM Lbones-ai ***@***.***> wrote:

> flixel: 4.6.2 4.6.3 4.8.0 4.8.1 [4.9.0]
> hscript: 2.3.0 [2.4.0]
> lime: 3.0.1 7.5.0 7.7.0 [7.9.0]
> openfl: 4.0.1 8.9.1 8.9.6 [9.1.0]
>
> I just updated haxe and all my installed haxelibs, now when I open the
> flixel debugger the text in "Stats" is broken:
> [image: immagine]
> <https://user-images.githubusercontent.com/84868536/122238125-2d100780-cec0-11eb-84bd-e5e6e28175f3.png>
>
> Also I'm getting those warnings:
> E:/HaxeToolkit/haxe/lib/hscript/2,4,0/hscript/Interp.hx:145: characters
> 8-18 : Warning : Std.is is deprecated. Use Std.isOfType instead.
> E:/HaxeToolkit/haxe/lib/hscript/2,4,0/hscript/Interp.hx:180: characters
> 8-18 : Warning : Std.is is deprecated. Use Std.isOfType instead.
> E:/HaxeToolkit/haxe/lib/hscript/2,4,0/hscript/Interp.hx:221: characters
> 8-18 : Warning : Std.is is deprecated. Use Std.isOfType instead.
> E:/HaxeToolkit/haxe/lib/hscript/2,4,0/hscript/Interp.hx:515: characters
> 8-18 : Warning : Std.is is deprecated. Use Std.isOfType instead.
> E:/HaxeToolkit/haxe/lib/hscript/2,4,0/hscript/Interp.hx:483: characters
> 37-56 : Warning : Std.is is deprecated. Use Std.isOfType instead.
> E:/HaxeToolkit/haxe/lib/hscript/2,4,0/hscript/Interp.hx:484: characters
> 31-47 : Warning : Std.is is deprecated. Use Std.isOfType instead.
> E:/HaxeToolkit/haxe/lib/hscript/2,4,0/hscript/Interp.hx:646: characters
> 10-42 : Warning : Std.is is deprecated. Use Std.isOfType instead.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/HaxeFlixel/flixel/discussions/2322>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAFUN6DBSPHPZLW7XO4MFETTTCYUBANCNFSM46ZSRIVA>
> .
>


-- 
www.fortressofdoors.com -- My design & business of game development blog

**Reply by Lbones-ai** [reply]
I'm using neko64 because compiles quick, I just tried with cpp and html5, the result it's the same

**Reply by larsiusprime** [reply]
I'm using:
flixel: 4.90
lime: 7.9.0
openfl: 9.10
haxe 4.2.1

Here's my results:
`lime test windows -debug -v`
![image](https://user-images.githubusercontent.com/739064/122245717-94f82b00-ce8b-11eb-9c0d-c9b2713085fc.png)

`lime test hl -debug -v`
![image](https://user-images.githubusercontent.com/739064/122245801-a80afb00-ce8b-11eb-805d-3e353baa69cb.png)

`lime test html5 -debug -v`
![image](https://user-images.githubusercontent.com/739064/122248443-afcb9f00-ce8d-11eb-9eb0-e962cc295c3f.png)

So it looks to be broken on all targets for me as well.

**Comment by Gama11** [comment, marked as answer]
This is an OpenFL bug that's already fixed on the dev branch afaik, but not released.

**Comment by TheHENOOB** [comment]
By fixing all of those issues that you said you can either use temporarily the git version of OpenFL and hscript until the bug fixes get released on haxelib

`haxelib git openfl https://github.com/openfl/openfl`
`haxelib git hscript https://github.com/HaxeFoundation/hscript`

or just copy and paste the code of the commits to fix those bugs which is a little harder way

hscript deprecation warnings: https://github.com/HaxeFoundation/hscript/commit/36eb0fca8b4ad0836c7b787f65833331746feb11
falling letters on flixel debugger: https://github.com/openfl/openfl/commit/cda4454a50fa4ed1e88c2e309b589cd99a746e93
